### ISSUE TITLE
Using DDPG for Pong

### ISSUE THREAD

**Issue by option-greek** [pemami4911/deep-rl#6]
Hi Patrick,
    I'm trying to convert your ddpg pendulum code to solve pong. I made minimal changes like input pre-processing and modifying of input, output dimensions. Over a few iterations I notice that the paddle sticks to the bottom of the screen as the probabilities for UP action becomes negligible and DOWN becomes almost equal to one. Since the original sample is for continuous space problem, I'm guessing since I'm expecting discrete output (up or down), I missed out changing some part of the code. Could you kindly look at my code here and point me to the missing piece:
https://gist.github.com/option-greek/dfc9288d5811371f578b2f52dce29f0e

Thanks,
OG

**Comment by pemami4911** [comment]
Sorry for the late reply, I've been quite busy. Have you had any luck? For starters, DDPG uses a deterministic policy, so the output of the policy network are the values for continuous actions. You'd have to change the output layer of the policy network (e.g., a softmax layer) and have the value network output action-values for each discrete action. Basically, you'd want to do something more along the lines of [A3C](https://arxiv.org/pdf/1602.01783.pdf). Also, changing the noise process added to the actions from what I have in this code, and using BatchNorm, should help as well.

**Comment by option-greek** [comment]
I had success with a A3C sample where the actor and critic were predicting the action and values respectively. I'm just curious as to whether DDPG isn't adaptable to discrete space in some way :) My modified DDPG sample already uses a softmax and doesn't use the randomization: https://gist.github.com/option-greek/dfc9288d5811371f578b2f52dce29f0e

Are you aware of any papers that adapt DDPG to discrete space (or is this completely unnecessary in practice because other algorithms are already good at discrete actions) ?

Thanks,
OG

**Comment by pemami4911** [comment]
I have not seen any papers that modify DDPG for discrete spaces. It's development was mainly motivated by the desire to adapt DQN to continuous action spaces. If you were to modify it, you'd probably just end up with something very similar to A3C anyways!

**Comment by PeixianChen** [comment]
I'm trying use the ddpg to solve Pong and have a same problem. Could you tell me you solve this problem or not? Sorry, my English is poor, hope you can understand me ! Thank you!
